File: backend/internal/core/domain/Group.go
package domain

import "go.mongodb.org/mongo-driver/bson/primitive"

// swagger:model Group
type Group struct {
	ID          primitive.ObjectID `bson:"_id,omitempty" json:"id,omitempty"`
	Name        string             `bson:"name" json:"name"`
	Owner       User               `bson:"owner" json:"owner"`
	Permissions []Permission       `bson:"permissions" json:"permissions"`
	Active      bool               `bson:"active" json:"active"`
	Public      bool               `bson:"public" json:"public"`
}

File: frontend/src/app/types/group.ts
import { createMockUser, User } from './user';
import { Permission, PermissionDto } from './permission';

export interface GroupDto {
    name: string;
    owner: User;
    permissions: PermissionDto[];
    public: boolean;
}

export interface Group {
    id: string;
    name: string;
    owner: User;
    permissions: Permission[];
    active: boolean;
    public: boolean;
}

export function createMockGroup(): Group {
    return {
        id: '1',
        name: 'group',
        owner: createMockUser(),
        active: true,
        permissions: [],
        public: true,
    };
}
